fili-core/src/main/java/com/yahoo/bard/webservice/logging/RequestLog.java

```diff
@@ -397,7 +402,20 @@ public static String getId() {
         if (current.info == null) {
             current.init();
         }
-        return current.logId;
+        return current.idPrefix + current.logId;
+    }
+
+    /**
+     * Set id prefix.
+     *
+     * @param idPrefix  Prefix for queryId sent to druid
+     */
+    public static void setIdPrefix(String idPrefix) {
+        RequestLog current = RLOG.get();
+        if (current.info == null) {
+            current.init();
+        }
+        current.idPrefix = (idPrefix == null) ? "" : idPrefix;
```

Comment:
This isn't enough. If you look at line [157](https://github.com/yahoo/fili/pull/68/files#diff-79f5168b37854c3172bc2994b14d3168R157), you'll see that we are putting the id into `MDC`. That `MDC` is what allows us to put the id into every log line generated by a request.

As a result, if we want the prefix to _also_ show up in every log line (we do), then we need to make sure the `logId` and the `idPrefix` are added to `MDC`. 

So at the minimum, in the last line in this method, we should add the following:

```java
MDPC.put(ID_KEY, getId())
```

Or manually do the string concatentation if calling the getter in the setter gives you heebie-jeebies.

Although in all honesty, I feel like that opens the door to subtle bugs, if `init` gets called later and overwrites `MDC` with the prefix. 

Instead, I think that `setIdPrefix` should prepend the id to `logId`:

```java
current.logId = idPrefix + current.logId;
MDC.put(ID_KEY, current.logId);
```

That way, if `init` gets called again at some other random point, we don't have to worry about the prefix being dropped.